### PR TITLE
Add host update signal

### PR DESCRIPTION
**Pull request by ghost**
`post_save` publish host data (using `DCHostSerializer`) on Hermes for models:
- `DataCenterAsset`
- `CloudHost`
- `VirtualServer`

### CODE REVIEW

File: src/ralph/settings/base.py [hunk 1]
```diff
@@ -392,6 +393,9 @@ def os_env_true(var, default=''):
 ENABLE_HERMES_INTEGRATION = os_env_true('ENABLE_HERMES_INTEGRATION')
 HERMES = json.loads(os.environ.get('HERMES', '{}'))
 HERMES['ENABLED'] = ENABLE_HERMES_INTEGRATION
+# topic name where DC asset, cloud host, virtual server changes should be
+# announced
+HERMES_HOST_UPDATE_TOPIC_NAME = None
```
Comment (ar4s) [inline src/ralph/settings/base.py]: maybe `os.environ.get('HERMES_HOST_UPDATE_TOPIC_NAME', None)`?

File: src/ralph/api/serializers.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 from functools import reduce
 
 import reversion
+from django.conf import settings
```
Comment (mkurek) [inline src/ralph/api/serializers.py]: use `from rest_framework.settings import api_settings`

File: src/ralph/api/serializers.py [hunk 3]
```diff
@@ -128,6 +134,12 @@ def get_fields(self, *args, **kwargs):
         for field_name, field in fields.items():
             if not field.parent:
                 field.parent = self
+        if (
+            self.skip_url_when_no_request and
+            not self.context.get('request') and
+            settings.REST_FRAMEWORK['URL_FIELD_NAME'] in fields
```
Comment (mkurek) [inline src/ralph/api/serializers.py]: use `api_settings.URL_FIELD_NAME`

File: src/ralph/api/serializers.py [hunk 4]
```diff
@@ -128,6 +134,12 @@ def get_fields(self, *args, **kwargs):
...
+        ):
+            del fields['url']
```
Comment (mkurek) [inline src/ralph/api/serializers.py]: `del fields[api_settings.URL_FIELD_NAME]`

File: src/ralph/publishers.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+def _get_host_data(instance):
+    serializer = DCHostSerializer(instance=instance)
+    return serializer.data
+
+if settings.HERMES_HOST_UPDATE_TOPIC_NAME:
+    @pyhermes.publisher(
+        topic=settings.HERMES_HOST_UPDATE_TOPIC_NAME, auto_publish_result=True
+    )
+    def publish_host_update(data):
+        return data
```
Comment (mkurek) [inline src/ralph/publishers.py]: use `_get_host_data` here, not in each signal separately.

File: src/ralph/publishers.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
```
Comment (mkurek) [inline src/ralph/publishers.py]: imo `ralph/publishers.py` is not the best place for this. Could it be:
- defined not in main scope of ralph?
- gather all signals in one place?

Maybe new app for such kind of things?

File: src/ralph/publishers.py (hunk 5 above)
Comment (xliiv) [inline src/ralph/publishers.py]: ?

File: src/ralph/publishers.py (hunk 6 above)
Comment (ghost) [inline src/ralph/publishers.py]: `ralph/publishers/__init__.py`?
`ralph/publishers/publishers.py`?
all is bad, so recommend something, please :)

Comment (mkurek) [inline src/ralph/publishers.py]: so maybe in models in particular app? besides registering signal, wrap it as pyhermes publisher

File: src/ralph/settings/base.py [hunk 7]
```diff
@@ -264,7 +264,8 @@ def os_env_true(var, default=''):
     'DEFAULT_METADATA_CLASS': 'ralph.lib.api.utils.RalphApiMetadata',
     'DEFAULT_VERSIONING_CLASS': 'rest_framework.versioning.AcceptHeaderVersioning',  # noqa
     'DEFAULT_VERSION': 'v1',
-    'ALLOWED_VERSIONS': ('v1',)
+    'ALLOWED_VERSIONS': ('v1',),
+    'URL_FIELD_NAME': 'url',
```
Comment (mkurek) [inline src/ralph/settings/base.py]: no need for this anymore

File: src/ralph/data_center/tests/test_signals.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    def test_dc_asset_is_serialized_ok(self):
+        data = _get_host_data(self.dc_asset)
+        self.assertTrue(isinstance(data, dict))
+
+    def test_cloud_host_is_serialized_ok(self):
+        data = _get_host_data(self.cloud_host)
+        self.assertTrue(isinstance(data, dict))
+
+    def test_asset_is_serialized_ok(self):
+        data = _get_host_data(self.dc_asset)
```
Comment (szok) [inline src/ralph/data_center/tests/test_signals.py]: the same as test_dc_asset_is_serialized_ok :)

**Comment by mkurek** [comment]
:ok:

**Comment by szok** [comment]
:ok:

**Pull request merged**